### PR TITLE
feat(#656): [Pooja] fixed rendering demographic data in requested language

### PR DESCRIPTION
**Pull request by PoojaBabusing**
Rendering the demographic data in requested language

### CODE REVIEW

**Review by vijay151096 — APPROVED** [review]
Looks Good to me.

File: components/VcDetails.tsx [hunk 1]
```diff
@@ -352,6 +353,18 @@ function getLocalizedField(rawField: string | LocalizedField[]) {
   }
   try {
     const locales: LocalizedField[] = JSON.parse(JSON.stringify(rawField));
+    const supportedLanguages = Object.keys(SUPPORTED_LANGUAGES);
+    const currentLanguage = i18n.language;
+    if (locales.length > 1) {
```
Comment (vijay151096) [inline components/VcDetails.tsx:358]: Since We are using the same code snippet in both places, can we extract a method and use that

**Review by vijay151096 — APPROVED** [review]
Looks Good.

File: i18n.ts [hunk 2]
```diff
@@ -43,3 +48,55 @@ function getLanguageCode(code: string) {
...
+}
+
+// This method gets the value from iso-639-3 package, which contains key value pairs of three letter language codes[key] and two letter langugae code[value]. These values are according to iso standards.
+// The response recieved from the server is three letter language code and the value in the inji code base is two letter language code. Hence the conversion is done.
+function getThreeLetterLanguageCode(twoLetterLanguageCode) {
+  let threeLetterLanguageCode = Object.keys(iso6393To1).find(
+    (key) => iso6393To1[key] === twoLetterLanguageCode
+  );
+  if (!threeLetterLanguageCode) {
+    threeLetterLanguageCode = Object.keys(iso6393To2T).find(
```
Comment (tilak-puli) [inline i18n.ts:75]: why are we using two different standards?

File: i18n.ts [hunk 3]
```diff
@@ -43,3 +48,55 @@ function getLanguageCode(code: string) {
   const [language] = code.split('-');
   return language;
 }
+
+export function getVCDetailsForCurrentLanguage(locales) {
```
Comment (tilak-puli) [inline i18n.ts:52]: can this function be refactored?

Please separate the nested logic into following:

1. check if supportedLanguage has current Language
2. if it has, get VC Details for current Language
3. if it doesn't, return safe value(I assume its returning undefined now)

**Comment by ravikp** [comment]
Kindly change the target branch to 'develop' branch.

File: i18n.ts (hunk 2 above)
Comment (PoojaBabusing) [reply i18n.ts:75]: In the supported languages list we have three letter code and two letter language codes. In this method getThreeLetterLanguageCode we pass all the supported languages and generate the three letter code. iso6393To1 list has key as 3 letter language code and value as 2 letter language code. If we pass 3 letter language code to iso6393To1 it return undefined. iso6393To2B has key and value as 3 letter language code. Hence we are using two different standards.

**Comment by PoojaBabusing** [comment]
Removing this PR and created a new one with latest develop.

**Pull request closed without merging**